### DISCUSSION TITLE
BACnet protocol revision no.

### DISCUSSION THREAD

**Discussion by kazemiprince** [JoelBender/bacpypes#535] [General]
Dear Joel,

Would you please let me know which BACnet protocol revision is compatible with bacpypes?

**Comment by JoelBender** [comment]
As a client (like read/write property services) the intent is to be up through BACnet-2020, revision 22.  There is no exhaustive set of tests that I can point to (like BTL listing!) that would demonstrate that.  It would be the same as a server, but that's a far more involved set of behaviors so it's probably closer to revision 12.  If there is some specific set of services that it doesn't have that you need for an application let me know, finding time to get BACnet/SC working is at the top of my list.

**Reply by kazemiprince** [reply]
Actually I create a building controller and used BACpypes as BACnet driver and I decided to get BTL certificate for it. 
Do you think it can pass the test? 
I can see controller is working fine and passed my test with Yabe and other server applications but I'm not sure what they will do in BTL labs for testing.

**Comment by JoelBender** [comment]
When you submitted the controller you provided the BTL with a set of specifications for what your device supports, BACnet Interoperability Building Blocks, and the BTL will run the tests associated with them.  The tests are specified in the companion standard [ASHRAE 135.1 Method of Test for Conformance to BACnet](https://webstore.ansi.org/standards/ashrae/ansiashraestandard1352019).  If you only provided device address binding (Who-Is/I-Am) then I'm sure it'll pass!  More importantly, if you get feedback of a failed test please make an issue for it.

**Reply by kazemiprince** [reply]
You mean I can apply for Who-Is/I-Am test only? is it possible? 
with "BACnet-Test Suite" Controller passed who-is - iam test but I'm not sure if we can apply test only for this two test and get BTL certificate for it. can you clear the possibility of this?
